Extension/src/LanguageServer/client.ts

```diff
@@ -947,6 +1094,154 @@ export class DefaultClient implements Client {
         this.languageClient.sendNotification(RenameNotification, params);
     }
 
+    private getWorkspaceFolderSettings(workspaceFolderUri: vscode.Uri | undefined, settings: CppSettings, otherSettings: OtherSettings): WorkspaceFolderSettingsParams {
```

Comment:
Suggestion to move `getWorkspaceFolderSettings` to file "Extension/src/LanguageServer/settings.ts" to reduce code in client.ts.

Same for `getAllWorkspaceFolderSettings` and `getAllSettings`.